optimize/upgrade/src/main/java/io/camunda/optimize/upgrade/plan/factories/UpgradePlanFactory.java

```diff
@@ -7,9 +7,42 @@
  */
 package io.camunda.optimize.upgrade.plan.factories;
 
+import io.camunda.optimize.service.db.DatabaseClient;
+import io.camunda.optimize.service.db.schema.OptimizeIndexNameService;
+import io.camunda.optimize.upgrade.exception.UpgradeRuntimeException;
 import io.camunda.optimize.upgrade.plan.UpgradeExecutionDependencies;
 import io.camunda.optimize.upgrade.plan.UpgradePlan;
+import java.io.IOException;
 
 public interface UpgradePlanFactory {
   UpgradePlan createUpgradePlan(final UpgradeExecutionDependencies upgradeExecutionDependencies);
+
+  void logErrorMessage(final String message);
+
+  default boolean isC7LicenseDataPresent(
+      final DatabaseClient databaseClient, final OptimizeIndexNameService indexNameService) {
+    try {
+      return databaseClient.countWithoutPrefixWithExistsCheck(
+              indexNameService.getOptimizeIndexAliasForIndex("license"))
+          > 0;
+    } catch (IOException e) {
+      final String reason = "Was not able to determine existence of C7 data.";
+      logErrorMessage(reason);
+      throw new UpgradeRuntimeException(reason, e);
+    }
+  }
+
+  default void validateOptimizeModeAndFailIfC7DataPresent(
```

Comment:
can/do we have an issue to remove this logic from this PR after the 8.6 release?

Reply:
Sure we can remove it when its no longer required.

Reply:
@grlimacan did you already have a follow up for this or shall I add one?

Reply:
I don't have one, but I am just wondering why this is no longer relevant for the next releases? Is it because there will simply be no upgrade factory that would try to perform an upgrade and therefore it's just technically impossible to try and upgrade a 3.14.X to an 8.x Optimize?